packages/api/src/external/commonwell/patient.ts

```diff
@@ -124,83 +116,78 @@ export async function create(
   });
 
   if (cwCreateEnabled) {
-    await setPatientDiscoveryStatus({
-      patientId: patient.id,
-      cxId: patient.cxId,
+    const discoveryRequestId = requestId ?? uuidv7();
+    const discoveryStartedAt = new Date();
+    const updatedPatient = await updatePatientDiscoveryStatus({
+      patient,
       status: "processing",
+      discoveryRequestId,
+      discoveryFacilityId: facilityId,
+      discoveryStartedAt,
+      discoveryRerunPdOnNewDemographics: rerunPdOnNewDemographics,
     });
 
-    // intentionally async
-    registerAndLinkPatientInCW(patient, facilityId, getOrgIdExcludeList, debug, requestId).catch(
-      processAsyncError(createContext)
-    );
+    const registerParams = {
+      patient: createAugmentedPatient(updatedPatient),
+      facilityId,
+      getOrgIdExcludeList,
+      rerunPdOnNewDemographics,
+      requestId: discoveryRequestId,
+      startedAt: discoveryStartedAt,
+      debug,
+      initiator,
+    };
+    if (callSynchronous) {
+      await registerAndLinkPatientInCW(registerParams);
+    } else {
+      // intentionally async
+      registerAndLinkPatientInCW(registerParams).catch(processAsyncError(createContext));
```

Comment:
I reviewed the code and it seems we don't need this. The `create` is called in two places, and in both of them we explicitly ask for synchronous or we can live with synchronous (`runInitialPatientDiscoveryAcrossHies`).

It would be great to remove this parameter, it exposes too much details about this function.

Reply:
Done.